table-api/src/main/java/io/deephaven/api/filter/FilterMatches.java

```diff
@@ -0,0 +1,52 @@
+package io.deephaven.api.filter;
+
+import io.deephaven.annotations.BuildableStyle;
+import io.deephaven.api.ColumnName;
+import io.deephaven.api.expression.Expression;
+import io.deephaven.api.literal.Literal;
+import org.immutables.value.Value.Default;
+import org.immutables.value.Value.Immutable;
+
+import java.util.List;
+
+@Immutable
+@BuildableStyle
+public abstract class FilterMatches extends FilterBase {
+
+    public static Builder builder() {
+        return ImmutableFilterMatches.builder();
+    }
+
+    public abstract Expression expression();
+
+    public abstract List<Literal> values();
```

Comment:
Okay - I'll expand this type b/c I agree. That said, there is room for a `QueryScope extends Expression` type in the future that we don't currently have.

Reply:
You need to support `FilterMatches` in some way, but it could be via equals, in, and case-insensitive String-specific variants of each (4 total).
I still think you need param support.

Reply:
I think we should follow-up later w/ https://github.com/deephaven/deephaven-core/issues/3784

Reply:
And https://github.com/deephaven/deephaven-core/issues/3785